pkg/antctl/raw/test/command.go

```diff
@@ -0,0 +1,522 @@
+
+func (k *k8sConnectivityParams) deleteDeployments(ctx context.Context, client k8sClientOperations) error {
+	k.Log("[%s] Deleting connectivity check deployments...", client.ClusterName())
+	client.DeleteDeployment(ctx, connectivityCheckNamespace, echoSameNodeDeploymentName, metav1.DeleteOptions{})
+	client.DeleteDeployment(ctx, connectivityCheckNamespace, echoOtherNodeDeploymentName, metav1.DeleteOptions{})
+	client.DeleteDeployment(ctx, connectivityCheckNamespace, clientDeploymentName, metav1.DeleteOptions{})
+	client.DeleteService(ctx, connectivityCheckNamespace, echoSameNodeDeploymentName, metav1.DeleteOptions{})
+	client.DeleteService(ctx, connectivityCheckNamespace, echoOtherNodeDeploymentName, metav1.DeleteOptions{})
+	client.DeleteNamespace(ctx, connectivityCheckNamespace, metav1.DeleteOptions{})
+
+	_, err := client.GetNamespace(ctx, connectivityCheckNamespace, metav1.GetOptions{})
+	if err == nil {
+		k.Log("[%s] Waiting for namespace %s to disappear", client.ClusterName(), connectivityCheckNamespace)
+		for err == nil {
+			time.Sleep(time.Second)
+			_, err = client.GetNamespace(ctx, connectivityCheckNamespace, metav1.GetOptions{})
+		}
+	}
+
+	return nil
+}
+
+func (k *k8sConnectivityParams) deploymentList() (srcList []string, dstList []string) {
+	srcList = []string{clientDeploymentName, echoSameNodeDeploymentName}
+
+	if !k.params.SingleNode {
+		dstList = append(dstList, echoOtherNodeDeploymentName)
+	}
+
+	return srcList, dstList
+}
+
+type deploymentClients struct {
+	source      k8sClientOperations
+	destination k8sClientOperations
+}
+
+func (d *deploymentClients) clients() []k8sClientOperations {
+	return []k8sClientOperations{d.source}
+}
+
+func (k *k8sConnectivityParams) initClients(ctx context.Context) (*deploymentClients, error) {
+	c := &deploymentClients{
+		source:      k.client,
+		destination: k.client,
+	}
+
+	daemonSet, err := k.client.GetDaemonSet(ctx, k.params.antreaNamespace, AgentDaemonSetName, metav1.GetOptions{})
+	if err != nil {
+		k.Log("Unable to determine status of Antrea DaemonSet.")
+		return nil, fmt.Errorf("Unable to determine status of antrea DaemonSet: %w", err)
+	}
+
+	if daemonSet.Status.DesiredNumberScheduled == 1 && !k.params.SingleNode {
+		k.Log("Single node environment detected, enabling single node connectivity test")
+		k.params.SingleNode = true
+	}
+
+	return c, nil
+}
+
+func (k *k8sConnectivityParams) deploy(ctx context.Context) error {
+	var srcDeploymentNeeded, dstDeploymentNeeded bool
+
+	if k.params.ForceDeploy {
+		if err := k.deleteDeployments(ctx, k.clients.source); err != nil {
+			return err
+		}
+	}
+
+	_, err := k.clients.source.GetNamespace(ctx, connectivityCheckNamespace, metav1.GetOptions{})
+	if err != nil {
+		srcDeploymentNeeded = true
+		dstDeploymentNeeded = true
+
+		k.Log("[%s] Creating namespace for connectivity check...", k.clients.source.ClusterName())
+		_, err = k.clients.source.CreateNamespace(ctx, connectivityCheckNamespace, metav1.CreateOptions{})
```

Comment:
could we add a random (short) suffix to the Namespace name?

Reply:
sure, implemented.